query_executor.go

```diff
@@ -63,6 +65,7 @@ func (q *queryExecutor) speculate(ctx context.Context, qry ExecutableQuery, sp S
 func (q *queryExecutor) executeQuery(qry ExecutableQuery) (*Iter, error) {
 	hostIter := q.policy.Pick(qry)
 
+	log.Debug().Msgf("host picked for query %s is %s", qry, hostIter().Info().connectAddress)
```

Comment:
Can add a log in `do()` as well where we either do not get a pool for a host or a connection for the pool

Reply:
Please check if `hostIter()` should be called just for the logs